wiring/src/spark_wiring_ble.cpp

```diff
@@ -2177,8 +2173,12 @@ ssize_t BleLocalDevice::getAdvertisingData(BleAdvertisingData* advertisingData)
     if (advertisingData == nullptr) {
         return SYSTEM_ERROR_INVALID_ARGUMENT;
     }
+    hal_ble_adv_params_t advParams = {};
+    advParams.size = sizeof(hal_ble_adv_params_t);
+    hal_ble_gap_get_advertising_parameters(&advParams, nullptr);
     advertisingData->clear();
-    size_t len = CHECK(hal_ble_gap_get_advertising_data(advertisingData->data(), BLE_MAX_ADV_DATA_LEN, nullptr));
+    CHECK_TRUE(advertisingData->resize((advParams.primary_phy == BLE_PHYS_CODED) ? BLE_MAX_ADV_DATA_LEN_EXT : BLE_MAX_ADV_DATA_LEN), SYSTEM_ERROR_NO_MEMORY);
```

Comment:
We should use `BLE_MAX_ADV_DATA_LEN_EXT_CONNECTABLE` here.

Reply:
See above (non-connectable application)